package/MDAnalysis/lib/picklable_file_io.py

```diff
@@ -0,0 +1,188 @@
+        self.seek(args[1])
+
+
+class BufferIOPicklable(io.BufferedReader):
+    """A picklable buffer object for read-only FilIO object.
+
+    This class provides a buffered :class:`io.BufferedReader` object that can be pickled. Note that this only works in reda mode.
+
+    Example
+    -------
+    ::
+        file = FileIOPicklable('filename')
+        buffer_wrapped = BufferIOPicklable(file)
+
+    See Also
+    ---------
+    FileIOPicklable
+    TextIOPicklable
+    """
+    def __init__(self, raw):
+        super().__init__(raw)
+        self.raw_class = raw.__class__
+
+    def __getstate__(self):
+        return self.raw_class, self.name, self.tell()
+
+    def __setstate__(self, args):
+        raw_class = args[0]
+        name = args[1]
+        raw = raw_class(name)
+        super().__init__(raw)
+        self.seek(args[2])
+
+
+class TextIOPicklable(io.TextIOWrapper):
+    """Character and line based picklable file-like object.
+
+    This class provides a file-like :class:`io.TextIOWrapper` object that can
+    be pickled. Note that this only works in read mode.
+
+    Example
+    -------
+    ::
+        file = FileIOPicklable('filename')
+        text_wrapped = TextIOPicklable(file)
+
+    See Also
+    ---------
+    FileIOPicklable
+    BufferIOPicklable
+    """
+    def __init__(self, raw):
+        super().__init__(raw)
+        self.raw_class = raw.__class__
+
+    def __getstate__(self):
+        return self.raw_class, self.name, self.tell()
+
+    def __setstate__(self, args):
+        raw_class = args[0]
+        name = args[1]
+        # raw_class is used for further expansion this functionality to
+        # GZip files, which also requires a text wrapper.
+        raw = raw_class(name)
+        super().__init__(raw)
+        self.seek(args[2])
+
+
+def pickle_open(name, mode='rt'):
+    """Open file and return a stream with pickle function implemented.
+
+    Built-in `io.open` function only returns an unpicklable file object.
+    In order to serialize a `MDAnalysis.core.Universe`, this function can
+    used to open trajectory/topology files--a object composition approach,
+    as opposed to class inheritance, which is more flexible and easier for
+    pickle implementation for new readers.
+
+    Note
+    ----
+    Should be only used with read mode.
+
+    Parameters
+    ----------
+    name : str;
+        a filename given a text or byte string.
+    mode: {'r', 'rt', 'rb'} (optional)
+        'r':  open for reading in text mode;
+        'rt': read in text mode (default);
+        'rb': read in binary mode;
+
+    Raises
+    ------
+    ValueError : if `mode` is not one of the allowed read modes
+
+    Returns
+    -------
+    stream-like object
+
+    See Also
+    --------
+    :func:`anyopen`
+    :func:`io.open`
+    """
+    if mode not in {'r', 'rt', 'rb'}:
+        raise ValueError("Only read mode ('r', 'rt', 'rb') files can be pickled.")
+    name = os.fspath(name)
```

Comment:
[os.fspath](https://docs.python.org/3/library/os.html#os.fspath) is only in Python ≥ 3.6. This is ok (I think @MDAnalysis/coredevs decided to not having to support 3.5) but you need to add an entry to CHANGELOG under changes that only Python >= 3.6 is supported. You must also change `setup.py` accordingly.

Reply:
In the whole discussion on Python 3 we had decided that Py 3.6 is the minimum supported version. So, the CHANGELOG _should_ already have contained this information but it didn’t so please add it. Thank you!